lib/resty/core/regex.lua

```diff
@@ -479,6 +480,71 @@ function ngx.re.find(subj, regex, opts, ctx, nth)
 end
 
 
+local function iterate_re_gmatch(self)
+    local compile_once = self._compile_once
+    local compiled = self._compiled
+    local subj = self._subj
+    local flags = self._flags
+    local pos = self._pos
+    -- The iterator is exhausted.
+    if not pos then
+        return nil
+    end
+
+    local rc = C.ngx_http_lua_ffi_exec_regex(compiled, flags, subj, #subj, pos)
+
+    if rc == PCRE_ERROR_NOMATCH then
+        if not compile_once then
+            destroy_compiled_regex(compiled)
+        end
+        self._pos = nil
+        return nil
+    end
+
+    if rc < 0 then
+        if not compile_once then
+            destroy_compiled_regex(compiled)
+        end
+        self._pos = nil
+        return nil, "pcre_exec() failed: " .. rc
+    end
+
+    if rc == 0 then
+        if band(flags, FLAG_DFA) == 0 then
+            self._pos = nil
+            return nil, "capture size too small"
+        end
+
+        rc = 1
+    end
+
+    self._pos = compiled.captures[1]
+    local res = collect_captures(compiled, rc, subj, flags)
```

Comment:
I was under the impression that this was done to avoid `return collect_captures()` from returning multiple values, but it seems `collect_captures` only returns one value anyways. Maybe we could spare ourselves the variable here and `return collect_capture()` as a tail call? Or any reason why not?